haskell/private/path_utils.bzl

```
@@ -318,3 +318,103 @@ def copy_all(ctx, srcs, dest):
             command = "mkdir -p {dest} && cp -L -R \"$@\" {dest}".format(dest = dest.path),
             arguments = [args],
         )
+
+def parse_pattern(pattern_str):
+    """Parses a string label pattern.
+
+    Args:
+      pattern_str: The pattern to parse.
+        Patterns are absolute labels in the local workspace. E.g.
+        `//some/package:some_target`. The following wild-cards are allowed:
+        `...`, `:all`, and `:*`. Also the `//some/package` shortcut is allowed.
+
+    Returns:
+      A struct of
+        package: A list of package path components. May end on the wildcard `...`.
+        target: The target name. None if the package ends on `...`. May be one
+          of the wildcards `all` or `*`.
+
+    """
+
+    # We only load targets in the local workspace anyway. So, it's never
+    # necessary to specify a workspace. Therefore, we don't allow it.
+    if pattern_str.startswith("@"):
+        fail("Invalid haskell_repl pattern. Patterns may not specify a workspace. They only apply to the current workspace")
+
+    # To keep things simple, all patterns have to be absolute.
+    if not pattern_str.startswith("//"):
+        fail("Invalid haskell_repl pattern. Patterns must start with //.")
+
+    # Separate package and target (if present).
+    package_target = pattern_str[2:].split(":", maxsplit = 2)
```

```suggestion
    package_target = pattern_str[2:].split(":", maxsplit = 1)
```

I don't understand the 2, is that a typo or is there a specific reason (if, please add a comment). In python, `maxsplit` gives the number of time the separator is looked for. So with `maxsplit = 2`, you'll get a maximum of two splits, so three tokens. I suppose that the bazel API followed this.